tests/unit/test_flist.py

```diff
@@ -82,30 +87,61 @@ def test_api_foldLeft(self):
     self.assertEqual(120.0, alist.foldLeft(1.0, lambda a, e: a * e))
 
   def test_api_foldLeft_with_short_stop(self):
+    """Testing API Cons.foldLeft with arg short_stop_func."""
     alist = fl(1, 2, 3, 4, 5)
-
-    result = alist.foldLeft(
-        0, lambda a, b: a + b,
-        short_stop_func=lambda v: v > 3)
+    f = lambda a, b: a + b # noqa
+    # test corner case where the func stops at identity
+    result = alist.foldLeft(0, f, short_stop_func=lambda v: v > -1)
+    self.assertIsInstance(result, Nil)
```

Comment:
Brilliant idea to add test case to cover corner case.  Two feedback for your reference. 
1. This is a good usage case in using [parameterized testing feature](https://docs.pytest.org/en/7.1.x/example/parametrize.html). 
2. Please keep the unit test case content to follow [AAA structure](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80)

Below is sample code for reference:
```python
@pytest.mark.parametrize(
    "indicator,short_stop_func,expected",
    [
        pytest.param(
            0, lambda v: v > 3, 6, id="FAST",
        ),
        pytest.param(
            0, lambda v: v > -1, flist.nil, id="corner_case",
        ),
    ])
def test_api_fold_left_with_short_stop(
    indicator, short_stop_func, expected):
  # Arrange
  alist = fl(1, 2, 3, 4, 5)

  # Action
  result = alist.foldLeft(
      0, lambda a, b: a + b,
      short_stop_func=short_stop_func)

  # Assert
  # The folding operation will stop at element 4.
  # So we have folding result as 0 + 1 + 2 + 3 = 6
  assert result == expected
```